src/main/java/com/artipie/YamlSettings.java

```diff
@@ -126,4 +136,35 @@ public Storage repoConfigsStorage() {
     private static boolean hasTypeFile(final YamlMapping cred) {
         return cred != null && "file".equals(cred.string("type"));
     }
+
+    /**
+     * Wrapping for auth cache.
+     * @since 0.22
+     */
+    private static final class CachedAuthWrap implements Authentication {
```

Comment:
I'd suggested to simplify the name of the class to `Cached` as `Wrap` is usually used for abstract classes, this more looks like cached decorator of `Authentication`